### DISCUSSION TITLE
How to add Search Params in createLazyFileRoute

### DISCUSSION THREAD

**Discussion by shareef99** [TanStack/router#1510] [Q&A] [answered]
Hello, 
Grateful for this wonderful router. 
I am trying it out on my new project.
I don't see any way to add `Search Params` in `createLazyFileRoute`.
If I have to add **Search Params** then I must use `createFileRoute` ?
I have read [search params docs](https://tanstack.com/router/latest/docs/framework/react/guide/search-params) but still not able to do it. 
Thanks

**Comment by SeanCassiere** [comment, marked as answer]
This has to do with how code-splitting works categorizing critical configuration and non-critical configuration.

Here's the doc page on it: https://tanstack.com/router/latest/docs/framework/react/guide/code-splitting#how-does-tanstack-router-split-code
